content/python/concepts/collections-module/terms/namedtuple/namedtuple.md

```diff
@@ -0,0 +1,60 @@
+---
+Title: 'namedtuple'
+Description: 'A tuple subclass with named fields'
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+Tags:
+  - 'Tuples'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+A **namedtuple** is a data type in the `collections` module. It is a [`tuple`](https://www.codecademy.com/resources/docs/python/tuples) subclass that allows you to create immutable objects with named fields. This improves code readability by providing meaningful names for each element and making the code more explicit.
+
+## Syntax
+
+```pseudo
+namedtuple(typename, field_names)
+```
+
+The field names are made by a list of strings such as ['a', 'b', 'c'].
+
+Field names can alternatively be a string of several elements separated by a comma or a space.
+
+## Additional Methods
+
+There are 3 specific `namedtuple` methods in addition to the standard methods inherited from tuples.
+All methods begin with an underscore to avoid conflicts with field names.
+
+-`._make(iterable)`: (Class method) Creates a new instance based on an existing sequence or iterable.
+
+-`._asdict()`: Returns a [`dict`](https://www.codecademy.com/resources/docs/python/dictionaries) with matching field names and values.
+
+-`._replace(kwarg)`: Returns a new instance with fields replaced by the value provided as a keyword argument.
+
+There are 2 specific attributes `._fields` and `_field_defaults` that will allow you respectively to list field names and to return a [`dict`](https://www.codecademy.com/resources/docs/python/dictionaries) with field names related to their default values.
```

Comment:
```suggestion
There are 3 specific `namedtuple` methods in addition to the standard methods inherited from tuples. All methods begin with an underscore to avoid conflicts with field names.

-`._make(iterable)`: (Class method) Creates a new instance based on an existing sequence or iterable.

-`._asdict()`: Returns a [`dict`](https://www.codecademy.com/resources/docs/python/dictionaries) with matching field names and values.

-`._replace(kwarg)`: Returns a new instance with fields replaced by the value provided as a keyword argument.

There are 2 specific attributes `._fields` and `_field_defaults` that will allow you respectively to list field names and to return a [`dict`](https://www.codecademy.com/resources/docs/python/dictionaries) with field names related to their default values.
```